packages/react-components/react-utilities-compat/library/src/memoize.ts

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable deprecation/deprecation */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Stylesheet } from './internals/Stylesheet';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+declare class WeakMap {
```

Comment:
this will probably explode if this would reach public API surface, with tslib setup for compat packages this should be removed as WeakMap is part of ts definitions

Reply:
@Hotell I need to port utility functions with zero changes so that we can't introduce any bugs for partners. I don't expect it to ever be exported.  Is the tslib WeakMap the exact same implementation?  If not, then I think I need to leave this in here.

Reply:
I understand what's at play here but  we might be forced to do some changes in order to not clash with v9 (not this case). I double-check the `.d.ts` rolluped bundle emit and it is not exposed so we are good with this one.

> is the tslib WeakMap the exact same implementation?

obviously it's not :D it's strongly typed with generics and other jazz 
<img width="852" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/ecb21a2e-0f05-45b9-ba31-1f8f1a63f7b5">

but because it's declared like this `declare class` it would clash with existing typescript lib definitions

<img width="724" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/035f8c1b-3be6-4490-bb5f-4b671e9c5f02">


if it would be declared as interface global override ( allowed behaviour in TS ) it  dangerous approach as it would affect whole application codebase , that's why I pointed this out. 

hope it's more clear now.